### ISSUE TITLE
Tflearn Trained model with pretrained embeddings on Android studio

### ISSUE THREAD

**Issue by sushiboo** [tflearn/tflearn#1088]
I have a CNN model for text classification which uses a pre-trained embedding of the glove. I have frozen that graph optimized for inference and using it on the android studio. The problem is when I try to pass the weights into the model for inference. I have a JSON file with the key-value pairs between the words and the embedding which I use to create an input of embeddings from the text that the user types in. I can already get the embeddings from the JSON file but when I try to feed it into the graph for inference, it gives me the following error:

java.lang.IllegalArgumentException: indices[0,3891] = -2 is not in [0, 7459)
    	 [[Node: EmbeddingLayer/embedding_lookup = Gather[Tindices=DT_INT32, Tparams=DT_FLOAT, _class=["loc:@EmbeddingLayer/W"], validate_indices=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](EmbeddingLayer/W/read, EmbeddingLayer/Cast)]]

This is the code I use for freezing the graph:

def freeze_graph(model_folder,output_graph="/Users/sushanlimbu/Desktop/MScProject/cnnModel/frozen_model.pb"):
    try:
        checkpoint = tf.train.get_checkpoint_state(model_folder)
        input_checkpoint = checkpoint.model_checkpoint_path
        print("[INFO] input_checkpoint:", input_checkpoint)
    except:
        input_checkpoint = model_folder
        print("[INFO] Model folder", model_folder) 
        
    output_node_names = "FullyConnected/Softmax"
    
    saver = tf.train.import_meta_graph(input_checkpoint + '.meta', clear_devices=True)
    
    graph = tf.get_default_graph()
    input_graph_def = graph.as_graph_def()
    
    with tf.Session() as sess:
        saver.restore(sess, input_checkpoint)

        output_graph_def = graph_util.remove_training_nodes(
            input_graph_def
        )
        # We use a built-in TF helper to export variables to constants
        output_graph_def = graph_util.convert_variables_to_constants(
            sess,                        # The session is used to retrieve the weights
            input_graph_def,             # The graph_def is used to retrieve the nodes 
            output_node_names.split(","), # The output node names are used to select the usefull nodes
        ) 
        with tf.gfile.GFile(output_graph, "wb") as f:
            f.write(output_graph_def.SerializeToString())
        print("%d ops in the final graph." % len(output_graph_def.node))

        print("[INFO] output_graph:",output_graph)
        print("[INFO] all done")

Here is the Java code I use to classify the input (array of float value containing the embeddings of all the words that the user types)

private void classify(float[] input){
        TFInference = new TensorFlowInferenceInterface(getAssets(), MODEL_FILE);

        TFInference.feed(INPUT_NODE, input, 1, input.length);
        TFInference.run(OUTPUT_NODES);
        float[] resu = new float[2];
        TFInference.fetch(OUTPUT_NODE, resu);
        tvResult.setText("Programmer: " + Float.toString(resu[0]) + "\n Construction" +  Float.toString(resu[1]));
        Log.e("Result: ", Float.toString(resu[0]));
    }

Here is the code I use for building the input (The result of this method is the one I feed into the graph for inference):

private float[] buildInput(List<String> words){
        maxLength = words.size();
        List<float[]> allInput = new ArrayList<float[]>();
        float[] input = new float[batchSize * maxLength]; 
            if (vocabMap == null) {
                Log.e("Error", "Empty Vocab");
            } else {
                int i = 0;
                for (String word : words) {
                    if (vocabMap.get(word) != null) {
                        float[] index = vocabMap.get(word);
                        allInput.add(index);
                        Log.e("i", Float.toString(input[i]));

                        for (float f : index){
                            Log.e("index", Float.toString(f));
                        }

                        i++;
                        if (i == maxLength)
                            break;
                    }

                }
            }
            for (float[] f : allInput){
                input = ArrayUtils.addAll(input, f);
            }
        return input;
    }

Thank you in advance 🍡
